client/lib/accept/dialog.jsx

```diff
@@ -15,31 +12,26 @@ import classnames from 'classnames';
  */
 import Dialog from 'components/dialog';
 
-const AcceptDialog = createReactClass( {
-	displayName: 'AcceptDialog',
+class AcceptDialog extends Component {
+	static displayName = 'AcceptDialog';
 
-	propTypes: {
+	static propTypes = {
 		translate: PropTypes.func,
 		message: PropTypes.node,
 		onClose: PropTypes.func.isRequired,
 		confirmButtonText: PropTypes.node,
 		cancelButtonText: PropTypes.node,
 		options: PropTypes.object,
-	},
+	};
 
-	getInitialState: function() {
-		return { isVisible: true };
-	},
+	state = { isVisible: true };
 
-	onClose: function( action ) {
+	onClose = action => {
+		this.setState( { isVisible: false } );
 		this.props.onClose( 'accept' === action );
```

Comment:
`onClose` is [provided by](https://github.com/Automattic/wp-calypso/blob/60ea236ee15be699b2e7a20fb668e5f920208dc5/client/lib/accept/index.js#L19-L29) `lib/accept/index.js` and unmounts this component, so it shouldn't come as a surprise that after this line has run, the component is no longer mounted. Moving `setState` before unmounting should fix this.